### PR TITLE
Initial decoder version with inter tools support

### PR DESCRIPTION
**Pull request by nijilk**
This is same as PR #405, but rebased to latest master and fixed few bugs.

### CODE REVIEW

**Comment by nijilk** [comment]
@joelsole @hassount 
Hi,
Can you please review the changes?
This is same as PR #405, rebased to latest master and fixed few issues.

File: Source/Lib/Decoder/Codec/EbDecUtils.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#ifndef EbDecUtils_h
+#define EbDecUtils_h
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static INLINE int get_relative_dist(OrderHintInfo *ps_order_hint_info,
+                                    int ref_hint, int order_hint)
+{
```
Comment (hassount) [inline Source/Lib/Decoder/Codec/EbDecUtils.h]: coding style

File: Source/Lib/Decoder/Codec/EbDecInterPrediction.c [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    EbPictureBufferDesc *recon_picture_buf = dec_hdl->cur_pic_buf[0]->ps_pic_buf;
+
+    /* scan through sub 8 blocks and see if anyof them is intra */
+    some_use_intra = 0;
+    const BlockSize bsize = part_info->mi->sb_type;
+    sub_x = part_info->subsampling_x;
+    sub_y = part_info->subsampling_y;
+    bool sub8_w = (block_size_wide[bsize] == 4) && sub_x;
+    bool sub8_h = (block_size_high[bsize] == 4) && sub_y;
+    if (sub8_h || sub8_w)
```
Comment (hassount) [inline Source/Lib/Decoder/Codec/EbDecInterPrediction.c]: coding style

File: Source/Lib/Decoder/Codec/EbDecInterPrediction.c [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+                    mi->interp_filters, is_intrabc);
+            }
+        }
+    }
+}
+
+void svtav1_predict_inter_block(
+    EbDecHandle *dec_hdl, PartitionInfo_t *part_info,
+    int32_t mi_row, int32_t mi_col, int32_t num_planes)
+{
```
Comment (hassount) [inline Source/Lib/Decoder/Codec/EbDecInterPrediction.c]: coding style

File: Source/Lib/Decoder/Codec/EbDecInterPrediction.c [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    return clamped_mv;
+}
+
+
+void svtav1_predict_inter_block_plane(
+    EbDecHandle *dec_hdl, PartitionInfo_t *part_info, int32_t plane,
+    int32_t build_for_obmc, int32_t mi_x, int32_t mi_y,
+    void *dst, int32_t dst_stride,
+    int32_t some_use_intra, int32_t bit_depth)
+{
```
Comment (hassount) [inline Source/Lib/Decoder/Codec/EbDecInterPrediction.c]: coding style

File: Source/Lib/Decoder/Codec/EbDecUtils.h (hunk 1 above)
Comment (nijilk) [reply Source/Lib/Decoder/Codec/EbDecUtils.h]: @hassount 
Following is the coding style, rt?

static void function(const int argument1,
                              const int argument2)
{
    do_something();
}

Comment (nijilk) [reply Source/Lib/Decoder/Codec/EbDecUtils.h]: @hassount 
Hi,
Following is the coding style, rt?
static void function(const int argument1,
                              const int argument2)
{
    do_something();
}

Comment (nijilk) [reply Source/Lib/Decoder/Codec/EbDecUtils.h]: @hassount 
Hi,
Following is the coding style, rt?
static void function(const int argument1,
                              const int argument2)
{
    do_something();
}

Comment (nijilk) [reply Source/Lib/Decoder/Codec/EbDecUtils.h]: @hassount 
Hi,
Following is the coding style, rt?
``` c
static void function(const int argument1,
                     const int argument2)
{
    do_something();
}
```

File: Source/Lib/Decoder/Codec/EbDecInterPrediction.c (hunk 2 above)
Comment (nijilk) [reply Source/Lib/Decoder/Codec/EbDecInterPrediction.c]: @hassount 
Thanks. Will fix and commit again.

File: Source/Lib/Decoder/Codec/EbDecUtils.h (hunk 1 above)
Comment (joelsole) [reply Source/Lib/Decoder/Codec/EbDecUtils.h]: @hassount , is the issue that there should be one line per argument?

Comment (hassount) [reply Source/Lib/Decoder/Codec/EbDecUtils.h]: nvm it's a multi-line statement

File: Source/Lib/Decoder/Codec/EbDecInterPrediction.c (hunk 3 above)
Comment (hassount) [reply Source/Lib/Decoder/Codec/EbDecInterPrediction.c]: nvm

File: Source/Lib/Decoder/Codec/EbDecInterPrediction.c (hunk 4 above)
Comment (hassount) [reply Source/Lib/Decoder/Codec/EbDecInterPrediction.c]: nvm

**Pull request merged**